src/main/scala/viper/silver/plugin/standard/termination/transformation/ExpTransformer.scala

```diff
@@ -75,6 +75,29 @@ trait ExpTransformer extends ErrorReporter {
           Seqn(Seq(right), Nil)()
       }
       Seqn(Seq(left, rightSCE), Nil)()
+    case (ase: AnySetExp, c) => ase match {
+      case AnySetCardinality(s) => Seqn(Seq(transformExp(s, c)), Nil)(ase.pos)
+      case AnySetUnion(l, r) => Seqn(Seq(transformExp(l, c), transformExp(r, c)), Nil)(ase.pos)
+      case AnySetIntersection(l, r) => Seqn(Seq(transformExp(l, c), transformExp(r, c)), Nil)(ase.pos)
+      case AnySetSubset(l ,r) => Seqn(Seq(transformExp(l, c), transformExp(r, c)), Nil)(ase.pos)
+      case AnySetMinus(l, r) => Seqn(Seq(transformExp(l, c), transformExp(r, c)), Nil)(ase.pos)
```

Comment:
A lot of these could actually be handled by the default implementation in transformUnknownExp; do we really need explicit cases for all of them?

Reply:
If the case split is really needed, then we could still save on the size of the match by matching on `AnySetUnExp` and `AnySetBinExp`:
```scala
case (ase: AnySetExp, c) => ase match {
    case u: AnySetUnExp => transformExp(u.op, c)
    case b: AnySetBinExp => Seqn(Seq(transformExp(b.l, c), transformExp(b.r, c)), Nil)(ase.pos)
    case unsupportedExp => ...
}
```

Reply:
that's a very good point. So far, `transformUnknownExp` has been overwritten to report an error but I'll use this implementation to handle most cases